packages/lit-html/rollup.config.js

```diff
@@ -13,36 +13,108 @@
  */
 
 import filesize from 'rollup-plugin-filesize';
-import { terser } from 'rollup-plugin-terser';
+import {terser} from 'rollup-plugin-terser';
+import copy from 'rollup-plugin-copy';
+import * as pathLib from 'path';
+import sourcemaps from 'rollup-plugin-sourcemaps';
+import replace from '@rollup/plugin-replace';
+
+// In CHECKSIZE mode we:
+// 1) Don't emit any files.
+// 2) Don't include copyright header comments.
+// 3) Don't include the "//# sourceMappingURL" comment.
+const CHECKSIZE = !!process.env.CHECKSIZE;
+if (CHECKSIZE) {
+  console.log('NOTE: In CHECKSIZE mode, no output!');
+}
+
+const skipBundleOutput = {
+  generateBundle(options, bundles) {
+    // Deleting all bundles from this object prevents them from being written,
+    // see https://rollupjs.org/guide/en/#generatebundle.
+    for (const name in bundles) {
+      delete bundles[name];
+    }
+  },
+};
+
+const entryPoints = ['lit-html', 'directives/if-defined'];
 
 export default {
-  input: 'lib/lit-html.js',
+  input: entryPoints.map((name) => `development/${name}.js`),
   output: {
-    file: 'lit-html.bundled.js',
+    dir: './',
     format: 'esm',
+    // Preserve existing module structure (e.g. preserve the "directives/"
+    // directory).
+    preserveModules: true,
+    sourcemap: !CHECKSIZE,
   },
   plugins: [
+    // Switch all DEV_MODE variable assignment values to false. Terser's dead
+    // code removal will then remove any blocks that are conditioned on this
+    // variable.
+    //
+    // Code in our development/ directory looks like this:
+    //
+    //   const DEV_MODE = true;
+    //   if (DEV_MODE) { // dev mode stuff }
+    //
+    // Note we want the transformation to `goog.define` syntax for Closure
+    // Compiler to be trivial, and that would look something like this:
+    //
+    //   const DEV_MODE = goog.define('lit-html.DEV_MODE', false);
+    //
+    // We can't use terser's compress.global_defs option, because it won't
+    // replace the value of a variable that is already defined in scope (see
+    // https://github.com/terser/terser#conditional-compilation). It seems to be
+    // designed assuming that you are _always_ using terser to set the def one
+    // way or another, so it's difficult to define a default in the source code
+    // itself.
+    replace({
+      'const DEV_MODE = true': 'const DEV_MODE = false',
```

Comment:
Does this take a regex? Maybe anchor it to the beginning of a line to speed it up?

Reply:
No, unfortunately does not seem to support regexes.